version_template.py

```diff
@@ -107,14 +102,29 @@ def buildNumber():
 
 
 def currentCommit():
+    import os
     from subprocess import check_output
-    return check_output('git log --pretty=oneline -n 1 -- $(pwd)', shell=True).split()[0]
+
+    return check_output(['git', 'log', '--pretty=oneline', '-n', '1', os.getcwd()]).split()[0]
 
 
 def dirty():
+    """
+    Returns True if the code has uncommitted diffs.
+
+    :return: bool
+    """
     from subprocess import call
-    return 0 != call('(git diff --exit-code '
-                     '&& git diff --cached --exit-code) > /dev/null', shell=True)
+    import os
+
+    with open(os.devnull, 'w')  as FNULL:
```

Comment:
Did you see https://docs.python.org/3/library/subprocess.html#subprocess.DEVNULL

Reply:
I did, but I didn't see it on the python 2 page.  I agree it's much more elegant.

Reply:
You're not using the Python 2 sub process. You're using the backport from Python 3 which has DEVNULL. So why not use it?